--- a/cmake/unitree_sdk2Targets.cmake
+++ b/cmake/unitree_sdk2Targets.cmake
@@ -48,7 +48,7 @@
 # Create imported target ddsc and ddscxx
 add_library(ddsc SHARED IMPORTED GLOBAL)
 set_target_properties(ddsc PROPERTIES
-    IMPORTED_LOCATION ${_IMPORT_PREFIX}/lib/libddsc.so
-    INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include"
+    IMPORTED_LOCATION ${_IMPORT_PREFIX}/lib/libddsc_unitree.so
+    INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/unitree/third-party"
     INTERFACE_LINK_LIBRARIES "Threads::Threads"
     IMPORTED_NO_SONAME TRUE)
@@ -55,7 +55,7 @@

 add_library(ddscxx SHARED IMPORTED GLOBAL)
 set_target_properties(ddscxx PROPERTIES
-    IMPORTED_LOCATION ${_IMPORT_PREFIX}/lib/libddscxx.so
-    INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include/ddscxx"
+    IMPORTED_LOCATION ${_IMPORT_PREFIX}/lib/libddscxx_unitree.so
+    INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/unitree/third-party/ddscxx"
     INTERFACE_LINK_LIBRARIES "Threads::Threads"
     IMPORTED_NO_SONAME TRUE)
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -1,22 +1,22 @@
 find_package(Threads REQUIRED)

-message(STATUS "Importing: ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddsc.so")
+message(STATUS "Importing: ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddsc_unitree.so")

 add_library(ddsc SHARED IMPORTED GLOBAL)
 set_target_properties(ddsc PROPERTIES
-        IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddsc.so)
+        IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddsc_unitree.so)
 set_target_properties(ddsc PROPERTIES IMPORTED_NO_SONAME TRUE)
 target_link_libraries(ddsc INTERFACE Threads::Threads)
 target_link_directories(ddsc INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR})
 target_include_directories(ddsc INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-        $<INSTALL_INTERFACE:include>)
+        $<INSTALL_INTERFACE:include/unitree/third-party>)

-message(STATUS "Importing: ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddscxx.so")
+message(STATUS "Importing: ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddscxx_unitree.so")

 add_library(ddscxx SHARED IMPORTED GLOBAL)
 set_target_properties(ddscxx PROPERTIES
-        IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddscxx.so)
+        IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libddscxx_unitree.so)
 set_target_properties(ddscxx PROPERTIES IMPORTED_NO_SONAME TRUE)
 target_link_libraries(ddscxx INTERFACE Threads::Threads)
 target_link_directories(ddscxx INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR})
@@ -23,8 +23,8 @@
 target_include_directories(ddscxx INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/ddscxx>
-        $<INSTALL_INTERFACE:include>
-        $<INSTALL_INTERFACE:include/ddscxx>)
+        $<INSTALL_INTERFACE:include/unitree/third-party>
+        $<INSTALL_INTERFACE:include/unitree/third-party/ddscxx>)

 ## Install targets
 file(GLOB DDS_LIBRARIES "${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/*")
@@ -32,4 +32,4 @@
 install(FILES ${DDS_LIBRARIES} DESTINATION ${CMAKE_INSTALL_LIBDIR})

 install(DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/include/
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/unitree/third-party)
